File: app/src/main/java/com/fabian/Fyi/MainActivity.java
package com.fabian.Fyi;

import android.os.Bundle;
import android.widget.ListView;

import androidx.appcompat.app.AppCompatActivity;

import com.fabian.Fyi.database.dao.MessageDao;
import com.fabian.Fyi.database.domain.Message;
import com.fabian.Fyi.adapter.MsgAdapter;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    MessageDao messageDao;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initDataBase();

        MsgAdapter msgAdapter = new MsgAdapter(MainActivity.this, R.layout.msg_item, messageDao.queryAll());
        ListView listview = findViewById(R.id.list);
        listview.setAdapter(msgAdapter);
    }

    public void initDataBase(){
        messageDao = new MessageDao(this);
        for (int i = 0; i < 10; i++) {
            messageDao.delete(i);
        }
    }

    public List<Message> initDate(){
        List<Message> messages = new ArrayList<>();
        for (int i = 0; i < 10; i++) {
            Message msg = new Message(i, "tag", "msg" + i, "detail", "2021-12-02 15:23:47");
            messages.add(msg);
        }
        return messages;
    }
}

File: app/src/main/java/com/fabian/fyi/MainActivity.kt
package com.fabian.fyi

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

//        final TextView textView = findViewById(R.id.clientId);
//        textView.setText("token");


//        textView.setOnClickListener(v -> {
//            ClipboardManager clipboard = (ClipboardManager) getApplicationContext().getSystemService(CLIPBOARD_SERVICE);
//            String text = textView.getText().toString();
//            ClipData myClip = ClipData.newPlainText("text", text);
//            clipboard.setPrimaryClip(myClip);
//            Toast.makeText(getApplicationContext(), text + " 已复制", Toast.LENGTH_SHORT).show();
//        });
    }
}